src/main/java/org/pitest/junit5/JUnit5TestUnitFinder.java

```diff
@@ -134,6 +158,14 @@ public void executionFinished(TestIdentifier testIdentifier, TestExecutionResult
             } else if (testIdentifier.isTest()) {
                 l.executionFinished(new Description(testIdentifier.getUniqueId(), testClass), true);
             }
+
+            if (serializeExecution) {
+                parentCoverageSerializers.remove(testIdentifier.getUniqueIdObject());

```

Comment:
We remove the lock from a map but don't unlock it? When is the lock released?

Reply:
The wall of text above should hopefully make it clear.
This is the lock for child tests, so it should not be locked at this point.